policy/policy-authority_test.go

```diff
@@ -176,3 +181,94 @@ func TestChallengesFor(t *testing.T) {
 		t.Error("Incorrect combinations returned")
 	}
 }
+
+func TestWillingToIssueWithWhitelist(t *testing.T) {
+	dbMap, cleanUp := paDBMap(t)
+	defer cleanUp()
+	pa, err := NewPolicyAuthorityImpl(dbMap, true)
+	test.AssertNotError(t, err, "Couldn't create policy implementation")
+	pa.DB.LoadRules(RuleSet{
+		Whitelist: []WhitelistRule{
+			{Host: "www.zombo.com"},
+		},
+	})
+	googID := core.AcmeIdentifier{
+		Type:  core.IdentifierDNS,
+		Value: "www.google.com",
+	}
+	zomboID := core.AcmeIdentifier{
+		Type:  core.IdentifierDNS,
+		Value: "www.zombo.com",
+	}
+
+	type listTestCase struct {
+		regID int64
+		id    core.AcmeIdentifier
+		err   error
+	}
+	testCases := []listTestCase{
+		{100, googID, ErrNotWhitelisted},
+		{-1, googID, nil},
```

Comment:
This test case is a little confusing because we expect that, in prod, goog would be blacklisted and would return ErrBlacklisted here. Probably worth a comment?

Also you might want to change the -1 here to `whitelistPartnerRegId`.